### DISCUSSION TITLE
Argdown-Markdown - the need to repeat header for every map

### DISCUSSION THREAD

**Discussion by antonostrovsky** [christianvoigt/argdown#252] [Q&A]
Hi Christian,

I greatly appreciate the value Argdown-Markdown syntax brings. However, my argdown headers are usually quite thick: I have title, subTitle, author, date, color, model, selection

It appears that every time I insert an argdown map I have to at least include the color, model and selection. I can't just include all these values at the top of the document and then have all maps below inherit these values.

Is there any chance I can define these values globally? For example, this is what I presently have to copy/paste into the head of each ```argdown ``` capsule:
```
color:
    colorScheme: iwanthue-colorblind-friendly
    tagColors:
        note: 3
        note2: 2
        pro: 4
        con: 
model:
    removeTagsFromText: true
selection:
    excludeDisconnected: false
```

**Comment by christianvoigt** [comment]
If you use `@argdown/cli` with the `argdown markdown` command, you can use an `argdown.config.json` file to define these settings for all code blocks in your Markdown file:

```json
{
    "color": {
        "colorScheme": "iwanthue-colorblind-friendly",
        "tagColors": {
            "note": 3,
            "note2": 2,
            "pro": 4
        }
    },
    "model": {
        "removeTagsFromText": true,
        "selection": {
            "excludeDisconnected": false
        }
    }
}
```

For the VSCode Markdown preview: VSCode does not tell Markdown-It what file is currently parsed (it only gives it the text content of the file). Without that information it is not possible for plugins like the Argdown plugin to know where to look for the `argdown.config.json` file. 

So much to my regret I could not implement this feature. [Take a look at this issue](https://github.com/microsoft/vscode/issues/84846#issuecomment-618285718) for more information.

**Comment by antonostrovsky** [comment]
Hello Christian,

Thank you for explaining. Could something like this https://github.com/marp-team/marp-vscode/pull/103/commits/a7f3c5eb37a1ee692c2e8a58dd374ae5a265c92b be used to resolve the file location? 

The use case would be that presently, without this in VSCode I have two options:

- Make do with unusual colours and VizJS layouts in VSCode, knowing that they will look different after being processed by argdown-cli
or
- include a large code block before each diagram.

Do you think this merits a messy fix, or better to wait for VS team to offer another way of locating the file?

I'll be happy to vote for the request if there is another case opened

**Comment by yhatt** [comment]
Hi, I'm an author of original issue https://github.com/microsoft/vscode/issues/84846.

Since VS Code 1.57, the current Markdown document has been exposed to markdown-it `env` object for rendering phase: https://github.com/microsoft/vscode/blob/b0ae2854927c19e4de54edd3890ec681f96476e5/extensions/markdown-language-features/src/markdownEngine.ts#L168. It means the workaround is no longer required to get the location where located the current Markdown.

I don't know if this change helps to resolve config of argdown. Just wanted to report the latest status 😉

**Reply by christianvoigt** [reply]
Thanks for the good news, I will open an issue and work on it the next time I update argdown-vscode

**Reply by uplight-dev** [reply]
any updates on this?
